src/components/Expense.js

```diff
@@ -1,30 +1,37 @@
 import { getUKFormattedDate, getUKFormattedEuros } from '../utils/helpers';
+import ConfirmDeleteModal from './ConfirmDeleteModal';
+
+const Expense = ({ expense, removeExpense }) => {
+  const { title, amount, date, id } = expense;
 
-const Expense = ({ title, amount, date, id, removeExpense }) => {
   return (
-    <li className="container gy-5 px-0 list-group-item">
-      <div className="row align-items-center">
-        <div className="col-9 col-lg-10">
-          <div className="row align-items-center">
-            <div className="col-lg-2 fs-5 fw-bold text-nowrap">
-              {getUKFormattedDate(date)}
-            </div>
-            <div className="col-lg-8 fs-5">{title}</div>
-            <div className="col-lg-2 fs-5 text-lg-end text-nowrap">
-              - {getUKFormattedEuros(amount)}
+    <>
+      <li className="container gy-5 px-0 list-group-item">
+        <div className="row align-items-center">
+          <div className="col-9 col-lg-10">
+            <div className="row align-items-center">
+              <div className="col-lg-2 fs-5 fw-bold text-nowrap">
+                {getUKFormattedDate(date)}
+              </div>
+              <div className="col-lg-8 fs-5">{title}</div>
+              <div className="col-lg-2 fs-5 text-lg-end text-nowrap">
+                - {getUKFormattedEuros(amount)}
+              </div>
             </div>
           </div>
+          <div className="col-3 col-lg-2 text-end">
+            <button
+              className="btn btn-danger"
+              data-bs-toggle="modal"
+              data-bs-target={`#modal${id}`}
+            >
+              Delete
+            </button>
+          </div>
         </div>
-        <div className="col-3 col-lg-2 text-end">
-          <button
-            className="btn btn-danger"
-            onClick={() => removeExpense({ title, amount, date, id })}
-          >
-            Delete
-          </button>
-        </div>
-      </div>
-    </li>
+      </li>
+      <ConfirmDeleteModal removeExpense={removeExpense} expense={expense} />
```

Comment:
do we render a full a full (hidden) modal for each expense?

if yes, then we could consider the power of React and only add the necessary markup to the dom when needed.

Reply:
refactored completely! now rendering only one modal when clicking on delete